appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java

```diff
@@ -264,21 +226,27 @@ private void postponeSendProof() {
   }
 
   private void handleCampaign() {
-    compositeDisposable.add(ReactiveNetwork.observeInternetConnectivity()
-        .subscribeOn(Schedulers.io())
-        .filter(hasInternet -> hasInternet)
-        .filter(__ -> this.campaignId == null)
-        .firstOrError()
-        .flatMap(__ -> campaignService.getCampaign())
-        .retryWhen(throwableObservable -> throwableObservable.toObservable()
-            .flatMap(throwable -> {
-              throwable.printStackTrace();
-              return ReactiveNetwork.observeInternetConnectivity();
-            })
-            .flatMap(this::retryIfNetworkAvailable)
-            .toFlowable(BackpressureStrategy.LATEST))
-        .doOnSuccess(this::processCampaign)
-        .subscribe());
+
+    if (campaignId == null) {
+      appcoinsClient.checkConnectivity(appcoinsClientResponsePing -> {
+
+        AppCoinsClientResponsePing pingResponse =
+            (AppCoinsClientResponsePing) appcoinsClientResponsePing;
+
+        if (pingResponse.HasConnection()) {
```

Comment:
What happens if we don't have network? Does the ping mechanism makes any retries? Or the process is stopped until the application is closed and opened again?